streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -240,8 +240,8 @@
                     STATE_DIR_DOC)
             .define(REPLICATION_FACTOR_CONFIG,
                     Type.INT,
-                    1,
-                    Importance.MEDIUM,
+                    3,
```

Comment:
my suggestion above was `min(3, brokers.size())` 

i dont like this cause it seems like magic, but it also addresses most peoples issues.